contrib/seeds/asmap.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# Copyright (c) 2013-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+import ipaddress
+
+# Convert a byte array to a bit array
+def DecodeBytes(byts):
+    return [(byt >> i) & 1 for byt in byts for i in range(8)]
+
+def DecodeBits(stream, bitpos, minval, bit_sizes):
```

Comment:
The code from here came from [testmap](https://github.com/sipa/asmap/blob/master/testmap.py) which doesn't come with a license. It looks like @sipa and @jamesob contributed to it, so to include it we would need you to sign off that it's acceptable to include this here under the MIT license.

Reply:
ACK MIT license from me.

Also, I'm currently rewriting the Python asmap code from scratch actually (to have a single class that can do creation of asmap files, decoding, lookup, diffing, have tests, ...), and it's close to being done.

Reply:
ACK MIT license

Reply:
Thank you!

> Also, I'm currently rewriting the Python asmap code from scratch actually (to have a single class that can do creation of asmap files, decoding, lookup, diffing, have tests, ...), and it's close to being done.

OK, we can wait for that, there's no hurry here. It would be nice to have this before 0.24 branch-off.

Reply:
> Also, I'm currently rewriting the Python asmap code from scratch actually (to have a single class that can do creation of asmap files, decoding, lookup, diffing, have tests, ...), and it's close to being done.

**/start slightly off-topic:**

Nice! Is the diffing for encoded mappings or just output from `asmap-rs`? Or would this even make `asmap-rs` obsolete? I'm in the process of making some improvements to UX there to show progress on RIPE data dump downloads and finding the ASN bottlenecks. I was also thinking of including a command there to compare non-encoded maps. I was going to use that to run some sort of monthly historical analysis over the past year to get an idea of how much the maps tend to change. Just don't want to duplicate effort.

Just some of the things to help get asmaps distributed with releases and enabled by default. @sipa, just let me know if you want to continue this chat off GitHub or on some other issue. :)

**/end slightly off-topic**